### ISSUE TITLE
BAB (Big-Ass Beam™), and beam weapon balance in general

### ISSUE THREAD

**Issue by Kaiser-Indrasil** [tsunamayo/Skywanderers#1776]
First things first, as a guy who couldn't get their hands on the copy of the game, I'm coming from the place of an unexperienced player. That said, the following thoughts are a product of a discussion on Discord and are my own outlook on the matter. With that out of the way, let's focus on...

The BAB (patent pending). Most of us fantasize about the idea of having a world-threatening weapon of mass destruction that can Thanos-snap capital ships out of existence with a single blow. While this sounds exciting, it's also extremely broken. Having played multiple sci-fi games that feature beam weaponry, including some PvP tournaments, I'm pretty confident with my assessment on the matter. But first, a screenshot - to both break up the block of text, and provide some reference.
![firefox_sasZBpVlA2](https://user-images.githubusercontent.com/63756057/79682838-86799980-821d-11ea-94b4-a439e5ca9da0.jpg)
That is a Sathanas. A 6 kilometer-long Juggernaut whose main means of planting the seeds of dread in everyone unlucky enough to face it is a front-mounted array of four beam cannons. In the game of Freespace 2 that features the behemoth, the balance is ensured by both limited range, and a painfully slow pace everything bigger than a bomber moves at. Obviously, that wouldn't - pardon the pun - fly in our game. With ships able to close the gap several times faster, we need other means of balancing. Ideally, the beam lasers should not "be all, end all" weapons system, but rather excel at particular scenarios and be too risky to deploy in others. Here's what I came up with:

- Introduce a spool-up time: 1s for small, 3s for medium, and 5s for big beam emitters. That will provide a window of opportunity to smaller, faster craft to escape from the bigger beam's path, or to increase the chance to dodge the smaller beam shots.

- Introduce two cooldown timers instead of one. It would work like this: The S (for "shorter") cooldown would dictate the minimal interval between the beam shots. However, if the pilot didn't wait for the L (for "longer") cooldown to pass, and chose to shoot the beam early, it would increase the energy consumption and/or heat generation of the beam emitter being used. That would increase the energy consumed and/or radiated exponentially, ensuring that even the most powerful ships would never be able to shoot more than a few times in a short succession. The energy consumption/radiation rates would return to their base values once the pilot waited for the L cooldown to pass.

![image](https://user-images.githubusercontent.com/63756057/79683323-238a0180-8221-11ea-9fc6-af22049a8236.png)
- Introduce a precision cone (bullet spray) for beam lasers. Beam lasers are hit-scan weapons, so they reach their target instantly - that's one advantage. One way to balance out that advantage would be to nerf the damage and call it a day, but that's not sexy, and would just provoke people to stack rows upon rows of beam emitters to compensate for low damage output. I say, leave the damage output high, and nerf the precision. That's where the above image comes in.
Before each shot and during the spool-up time, the beam emitter would pick a random angle within the cone, and stick to that vector throughout the entirety of the firing cycle. That way, even if the pilot was dead-accurate, they could still miss a shot against a small target and/or from too far away. Big beams would still be dangerous against a big target that mostly falls into the firing cone though. And, even more so, against smaller targets that didn't get out of the way soon enough. The smaller the beam emitter, the better the precision and the shorter the cooldowns... but smaller damage output and shorter range, too.

- Speaking of range, we can expand the tactical options by introducing the falloff range, at which the damage output would drop to 0%, and the optimal range, at which the damage output would still be sitting at 100%. The pilot would have an option to finish off a big nearly-dead target just inside their falloff range but would have to move in the optimal range to ensure maximum damage potential during regular engagements. Obviously, the numbers should go up with size, up to 2kms falloff and 1km optimal range for the BAB.*

I have a pretty strong opinion on the matter of beam lasers. While I love the idea of BAB, I cannot forget about countless times a broken game balance brought frustration instead of joy to my gameplay. Hopefully these mechanics strike a nice compromise and find a suitable gameplay niche for beam lasers.

*requires further balancing

**Comment by Kaiser-Indrasil** [comment]
![image](https://user-images.githubusercontent.com/63756057/79684331-5b487780-8228-11ea-9c34-7263382754c9.png)
The "accuracy vs precision" image was transparent. Woops.

**Comment by tsunamayo** [comment]
Thanks for the suggestion, indeed I can see why people love the idea of making huge weapon systems...
For now I have no plan for this, but if I have time again I will bring back the merging beam tech. That will make big-ass beam possible in a way... By design they will have to be mounted on the ship itself which will limit their use against smaller craft. (or the turret will be painfully slow as heavy...)
Cheers

**Comment by Kaiser-Indrasil** [comment]
> By design they will have to be mounted on the ship itself which will limit their use against smaller craft. (or the turret will be painfully slow as heavy...)
Exactly. AND, it could lower the ship's turning rate while firing it to make sure the pilot has next to no chance hitting the smaller craft.
